src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,235 @@
+        return addDataFileMessage;
+    }
+
+    /**
+     * Returns UI message when a task is marked as done.
+     *
+     * @param taskIndex Index of task marked as done
+     * @param currList Task list with updated list of tasks.
+     * @return UI message for done task.
+     */
+    public String markAsDone(int taskIndex, TaskList currList) {
+        String markedDoneMessage = this.lines + "\n" + "     Nice! I've marked this task as done:"
+                + "\n" + "       " + currList.get(taskIndex) + "\n" + this.lines;
+        return markedDoneMessage;
+    }
+
+    /**
+     * Returns UI message when a task has been added to the task list.
+     *
+     * @param newTask Task added.
+     * @param currList Task list with updated list of tasks.
+     * @return UI message for task added.
+     */
+    public String addTask(Task newTask, TaskList currList) {
+        String taskAddedUi = this.lines + "\n" + "     Got it. I've added this task:" + "\n" + "       "
+                + newTask + "\n" + "     Now you have " + String.valueOf(currList.getNumTask())
+                + " task(s) in the list." + "\n" + this.lines;
+        return taskAddedUi;
+    }
+
+    /**
+     * Returns UI message to list out tasks in the current task list.
+     *
+     * @param currList Task list with updated list of tasks.
+     * @return UI message of task list.
+     */
+    public String listItems(TaskList currList) {
+        String listInString = this.lines + "\n" + "     Here are the tasks in your list:" + "\n"
+                + currList.listItems() + this.lines + "\n";
+        return listInString;
+    }
+
+    /**
+     * Returns UI message when a task is removed.
+     *
+     * @param removeTask Removed task.
+     * @param currList Task list with updated list of tasks.
+     * @return UI message of removed task.
+     */
+    public String deleteTask(Task removeTask, TaskList currList) {
+        String deleteMessage = this.lines + "\n" + "     Alright, the following task has been removed" + "\n"
+                + "       " + removeTask + "\n" + "     Now you have " + String.valueOf(currList.getNumTask())
+                + " task(s) in the list." + "\n" + this.lines;
+        return deleteMessage;
+    }
+
+    /**
+     * Returns UI message when users input an invalid command.
+     *
+     * @param ex Relevant invalid command message.
+     */
+    public static void commandError(InvalidCommand ex) {
+        System.out.println("    ____________________________________________________________");
+        System.out.println("     " + ex);
+        System.out.println("    ____________________________________________________________");
+    }
+
+    /**
+     * Returns UI message when users close the bot.
+     *
+     * @return UI message for goodbye.
+     */
+    public String goodBye() {
+        String endGreeting = this.lines + "\n" + "     Bye. Hope to see you again soon!\n" + this.lines;
```

Comment:
endGreeting can be misleading as part of the greeting message, can consider using farewell or closingMessage to be more explicit.